content/docs/reference/cryptography.md

```diff
@@ -0,0 +1,70 @@
++++
+title = "Cryptography"
+description = "Cryptographic design"
+date = 2021-05-01T08:00:00+00:00
+updated = 2021-05-01T08:00:00+00:00
+draft = false
+weight = 10
+sort_by = "weight"
+template = "docs/page.html"
+
+[extra]
+lead = "How Biscuit uses cryptography for attenuation."
+toc = true
+top = false
++++
+
+Biscuit uses public key cryptography to build its tokens: the private key is
+required to create a token, and must be kept safe. The public key can be
+distributed, and is needed to verify a token.
+
+Specifically, it uses the [Ed25519 algorithm](https://en.wikipedia.org/wiki/EdDSA).
+
+A public key signature proves that the signed data has not been modified.
+So how does Biscuit implement attenuation, where a new valid token can be
+created from an existing one?
+
+The token uses a scheme inspired from public key infrastructure, like TLS certificates.
+It is made of a list of blocks, each of them containing data, a signature
+and a public key. When creating the token, we generate a random key pair.
+The root private key is used to sign the data and the new public key.
+The token then contains two blocks:
+- first block:
+  - data
+  - new public key
+  - signature
+- proof block:
+  - new private key
+
+<img src="/img/authority.svg" style="width: 100%" />
+
+To verify that token, we need to know the root public key. With that key,
+we check the signature of the first block. Then we take the public key
+from the first block, and verify that it matches the private key from
+the proof block.
+
+## Attenuation
+
+If we have a valid token, to create a new one, we remove the last block
```

Comment:
In practice we don't modify a token when we attenuate it, we create a new one, so I find "remove" a bit misleading.

Something like 

> To create an attenuated token from an existing one, we take the blocks, add a new one with the new payload and a public key from a newly generated keypair. The new proof block is the private key from the newly generated keypair. The old proof block is _not_ included in the token